sqs/src/main/scala/akka/stream/alpakka/sqs/SqsModel.scala

```diff
@@ -125,82 +126,78 @@ object MessageAction {
 }
 
 /**
- * Additional identifiers for FIFO messsages
+ * Messages returned by a SqsPublishFlow.
  *
- * @see [https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-additional-fifo-queue-identifiers.html]
+ * @param responseMetadata the SQS response metadata (AWS request ID, ...)
+ * @param metadata the SQS result metadata.
+ *                 This is either the SQS response itself or the SQS result entry in case of batching
+ * @param request the SQS send message request
  */
-final class FifoMessageIdentifiers @InternalApi private[sqs] (
-    val sequenceNumber: String,
-    val messageGroupId: String,
-    val messageDeduplicationId: Option[String]
+final class SqsPublishResult[+T <: SdkPojo] @InternalApi private[sqs] (
+    val responseMetadata: SqsResponseMetadata,
+    val metadata: T,
+    val request: SendMessageRequest
 ) {
 
   /** Java API */
-  def getSequenceNumber: String = sequenceNumber
+  def getResponseMetadata: SqsResponseMetadata = responseMetadata
 
   /** Java API */
-  def getMessageGroupId: String = messageGroupId
+  def getMetadata: T = metadata
 
   /** Java API */
-  def getMessageDeduplicationId: java.util.Optional[String] = messageDeduplicationId.asJava
+  def getRequest: SendMessageRequest = request
 
-  override def toString: String =
-    s"FifoMessageIdentifiers(sequenceNumber=$sequenceNumber, messageGroupId=$messageGroupId, messageDeduplicationId=$messageDeduplicationId)"
+  override def toString =
+    s"""SqsPublishResult(responseMetadata=$responseMetadata,metadata=$metadata,request=$request)"""
 
   override def equals(other: Any): Boolean = other match {
-    case that: FifoMessageIdentifiers =>
-      java.util.Objects.equals(this.sequenceNumber, that.sequenceNumber) &&
-      java.util.Objects.equals(this.messageGroupId, that.messageGroupId) &&
-      java.util.Objects.equals(this.messageDeduplicationId, that.messageDeduplicationId)
+    case that: SqsPublishResult[T] =>
+      java.util.Objects.equals(this.responseMetadata, that.responseMetadata) &&
+      java.util.Objects.equals(this.metadata, that.metadata)
```

Comment:
Good catch. Should have triggered an error in the test but I made an error there.

I'm still wondering why in `alpakka` we didn't go the `akka-http` way for the model: Java abstract classes/interface with the java api in a `javadsl` package and scala case classes in `scaladsl` extending the java api. We'll then not have to case of the `toString` and `equal`. Is it because of `copy` ?

Reply:
Yes. Copy's generated default parameters make it hard to evolve case classes in a binary-compatible way.